cli/azd/pkg/commands/pipeline/azdo_provider.go

```diff
@@ -7,65 +7,528 @@ import (
+		}
+	default:
+		panic(fmt.Sprintf("unexpected selection index %d", idx))
+	}
+	return projectName, projectId, newProject, nil
+}
+
 // configureGitRemote set up or create the git project and git remote
-func (p *AzdoHubScmProvider) configureGitRemote(ctx context.Context, repoPath string, remoteName string, console input.Console) (string, error) {
-	return "remoteUrl", errors.New("not implemented")
+func (p *AzdoScmProvider) configureGitRemote(ctx context.Context, repoPath string, remoteName string, console input.Console) (string, error) {
+	projectName, projectId, newProject, err := p.ensureProjectExists(ctx, console)
+	if err != nil {
+		return "", err
+	}
+
+	repoDetails := p.getRepoDetails()
+	repoDetails.projectName = projectName
+	repoDetails.projectId = projectId
+
+	err = p.saveEnvironmentConfig(azdo.AzDoEnvironmentProjectIdName, projectId)
+	if err != nil {
+		return "", fmt.Errorf("error saving project id to environment %w", err)
+	}
+
+	err = p.saveEnvironmentConfig(azdo.AzDoEnvironmentProjectName, projectName)
+	if err != nil {
+		return "", fmt.Errorf("error saving project name to environment %w", err)
+	}
+	var remoteUrl string
+
+	if !newProject {
+		remoteUrl, err = p.promptForAzdoRepository(ctx, console)
+		if err != nil {
+			return "", err
+		}
+	} else {
+		remoteUrl, err = p.getDefaultRepoRemote(ctx, projectName, projectId, console)
+		if err != nil {
+			return "", err
+		}
+	}
+	return remoteUrl, nil
+}
+
+// returns the git remote for a newly created repo that is part of a newly created AzDo project
+func (p *AzdoScmProvider) getDefaultRepoRemote(ctx context.Context, projectName string, projectId string, console input.Console) (string, error) {
+	connection, err := p.getAzdoConnection(ctx)
+	if err != nil {
+		return "", err
+	}
+	repo, err := azdo.GetDefaultGitRepositoriesInProject(ctx, projectName, connection)
+	if err != nil {
+		return "", err
+	}
+
+	err = p.StoreRepoDetails(ctx, repo)
+	if err != nil {
+		return "", err
+	}
+
+	message := fmt.Sprintf("using default repo (%s) in newly created project(%s)", *repo.Name, projectName)
+	console.Message(ctx, message)
+
+	return *repo.RemoteUrl, nil
+}
+
+// prompt the user to select azdo repo or create a new one
+func (p *AzdoScmProvider) promptForAzdoRepository(ctx context.Context, console input.Console) (string, error) {
+	var remoteUrl string
+	// There are a few ways to configure the remote so offer a choice to the user.
+	idx, err := console.Select(ctx, input.ConsoleOptions{
+		Message: fmt.Sprintf("How would you like to configure your remote? (Organization: %s)", p.repoDetails.projectName),
+		Options: []string{
+			"Select an existing Azure DevOps Repository",
+			"Create a new private Azure DevOps Repository",
+		},
+		DefaultValue: "Create a new private Azure DevOps Repository",
+	})
+
+	if err != nil {
+		return "", fmt.Errorf("prompting for remote configuration type: %w", err)
+	}
+
+	switch idx {
+	// Select from an existing Azure Devops project
```

Comment:
NIT: the comments for these options are sort of out of date.

Reply:
https://github.com/Azure/azure-dev/issues/756